cli/Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -134,6 +109,122 @@ public final class ServerAuthenticationController: ServerAuthenticationControlli
+                ):
+                    if legacyToken != nil {
+                        upToDateToken = token
+                    } else if let accessToken {
+                        // We consider a token to be expired if the expiration date is in the past or 30 seconds from now
+                        let now = Date.now()
+                        let expiresIn = accessToken.expiryDate
+                            .timeIntervalSince(now)
+                        let refresh = expiresIn < 30 || forceRefresh
+
+                        #if canImport(TuistSupport)
+                            Logger.current.debug(
+                                "Access token expires in less than \(expiresIn) seconds. Renewing..."
+                            )
+                        #endif
+                        if refresh {
+                            guard let refreshToken else {
+                                throw ServerClientAuthenticationError.notAuthenticated
+                            }
+                            #if canImport(TuistSupport)
+                                Logger.current.debug("Refreshing access token for \(serverURL)")
+                            #endif
+                            let tokens = try await refreshTokens(
+                                serverURL: serverURL, refreshToken: refreshToken
+                            )
+                            #if canImport(TuistSupport)
+                                Logger.current.debug("Access token refreshed for \(serverURL)")
+                            #endif
+                            upToDateToken = .user(
+                                legacyToken: nil,
+                                accessToken: try Self.parseJWT(tokens.accessToken),
+                                refreshToken: try Self.parseJWT(tokens.refreshToken)
+                            )
+                            expiresAt = try ServerAuthenticationController.parseJWT(tokens.accessToken)
+                                .expiryDate
+                        } else {
+                            upToDateToken = .user(legacyToken: nil, accessToken: accessToken, refreshToken: refreshToken)
+                            expiresAt = accessToken.expiryDate
+                        }
+                    } else {
+                        throw ServerClientAuthenticationError.notAuthenticated
+                    }
+                }
+                return (value: upToDateToken, expiresAt: expiresAt)
+            }
+    }
+
+    private func fetchTokenFromStore(serverURL: URL) async throws -> AuthenticationToken? {
+        var credentials: ServerCredentials? = try await credentialsStore.read(
+            serverURL: serverURL
+        )
+        if isTuistDevURL(serverURL), credentials == nil {
+            credentials = try await credentialsStore.read(
+                serverURL: URL(string: "https://cloud.tuist.io")!
+            )
```

Comment:
we should probably also migrate to using the `tuist.dev` in the credentials store, too, and eventually remove this one. A bit out-of-scope when it comes to this PR, though.